File: src/main/java/sc/iview/viewing/StopAnimation.java
package sc.iview.viewing;

import org.scijava.plugin.Parameter;
import org.scijava.plugin.Plugin;

import sc.iview.SciView;
import sc.iview.SciViewService;

import org.scijava.command.Command;

@Plugin(type = Command.class, 
		menuPath = "SciView>View>Stop Animation")
public class StopAnimation  implements Command {

	@Parameter
	private SciViewService sceneryService;

	@Override
	public void run() {
		Thread animator = sceneryService.getActiveSciView().getAnimationThread();
		if( animator != null ) {
			animator.stop();
			sceneryService.getActiveSciView().setAnimationThread( null );
		}

	}

}

File: src/main/kotlin/sc/iview/commands/view/StopAnimation.kt
package sc.iview.commands.view

import org.scijava.command.Command
import org.scijava.plugin.Menu
import org.scijava.plugin.Parameter
import org.scijava.plugin.Plugin
import sc.iview.SciView
import sc.iview.commands.MenuWeights.VIEW
import sc.iview.commands.MenuWeights.VIEW_STOP_ANIMATION

/**
 * Command to stop all current animations
 *
 * @author Kyle Harrington
 */
@Plugin(type = Command::class, menuRoot = "SciView", menu = [Menu(label = "View", weight = VIEW), Menu(label = "Stop Animation", weight = VIEW_STOP_ANIMATION)])
class StopAnimation : Command {
    @Parameter
    private lateinit var sciView: SciView

    override fun run() {
        sciView.stopAnimation()
    }
}